test/solo/notifications.test.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+// oboe's notifier will send all log messages
+process.env.APPOPTICS_DEBUG_LEVEL = 6;
+
+const ao = require('../..');
+const notifications = ao.notifications;
+
+const sources = {
+  oboe: {
+    config: 0,
+    'keep-alive': 0,
+    logging: 0,
+    'error-status': 0,
+  },
+  collector: {
+    'remote-config': 0,
+    'remote-warning': 0,
+  },
+  notifier: {
+    error: 0,
+  },
+  other: [],
+};
+
+const logLevelCounts = {
+  fatal: 0,
+  error: 0,
+  warn: 0,
+  info: 0,
+  low: 0,
+  medium: 0,
+  high: 0,
+  other: [],
+};
+
+const messages = [];
+
+notifications.on('message', function (msg) {
+  messages.push(msg);
+})
+
+const expect = require('chai').expect;
+
+describe('notification function tests (long tests)', function () {
```

Comment:
where are the messages coming from?

Reply:
lines 158, 162, or 172 of `lib/notifications.js` where it executes `this.emit()`.

events are essentially the same thing as callbacks except there can be multiple listeners on an event while a callback is a singleton.

Reply:
but where do the messages get created that are emitted? I think I'm missing something in the way this test is setup.

Reply:
oh, i see. it counts on getting the messages from the production collector. one of the reasons these needed to be standalone tests is that they set the environment to use the production collector and also set oboe's log level to 6 in order to assure that the right messages are received. if things change then the checks for the test might need to change but that should be obvious then.